packages/editor-ui/src/components/NDVDraggablePanels.vue

```diff
@@ -7,28 +7,57 @@
 			<slot name="output"></slot>
 		</div>
 		<div :class="$style.mainPanel" :style="mainPanelStyles">
-			<div :class="$style.dragButtonContainer" @click="close">
-				<PanelDragButton
-					:class="{ [$style.draggable]: true, [$style.visible]: isDragging, [$style['double-width']]: hasDoubleWidth }"
-					v-if="!hideInputAndOutput && isDraggable"
-					:canMoveLeft="canMoveLeft"
-					:canMoveRight="canMoveRight"
-					@dragstart="onDragStart"
-					@drag="onDrag"
-					@dragend="onDragEnd"
-				/>
-			</div>
-			<slot name="main"></slot>
+			<n8n-resize-wrapper
+				:isResizingEnabled="true"
+				:width="relativeWidthToPx(mainPanelDimensions.relativeWidth)"
+				:minWidth="MINIMUM_INPUT_PANEL_WIDTH"
+				:gridSize="20"
+				@resize="onResize"
+				@resizestart="onResizeStart"
+				@resizeend="onResizeEnd"
+				:supportedDirections="supportedResizeDirections"
+			>
+				<div :class="$style.dragButtonContainer">
+					<PanelDragButton
+						:class="{ [$style.draggable]: true, [$style.visible]: isDragging }"
+						:canMoveLeft="canMoveLeft"
+						:canMoveRight="canMoveRight"
+						v-if="!hideInputAndOutput && isDraggable"
+						@dragstart="onDragStart"
+						@drag="onDrag"
+						@dragend="onDragEnd"
+					/>
+				</div>
+				<div :class="{ [$style.mainPanelInner]: true, [$style.dragging]: isDragging }">
+					<slot name="main" />
+				</div>
+			</n8n-resize-wrapper>
 		</div>
 	</div>
 </template>
 
 <script lang="ts">
 import Vue from 'vue';
 import PanelDragButton from './PanelDragButton.vue';
+import { get } from 'lodash';
```

Comment:
we should try and avoid lodash get and set because they lose typing + other.. we want to get rid of them next.. waiting on optional chaining so we don't lose typing.. you don't have to change it here but just heads up for future changes..
https://linear.app/n8n/issue/N8N-4533

Reply:
Agree. Optional chaining would also be my go-to for a case like this.